File: scratch/fix_admin_users_view.js
const fs = require("fs");
const adminPanelPath = "frontend/src/pages/AdminPanel.tsx";
const headPath = "scratch/AdminPanel_head_utf8.tsx";

let adminContent = fs.readFileSync(adminPanelPath, "utf-8");
let headContent = fs.readFileSync(headPath, "utf-8");

const startIdx = headContent.indexOf("{activeView === \x27users\x27 && (");
const endIdx = headContent.indexOf("{activeView === \x27pending\x27");

if (startIdx === -1 || endIdx === -1) {
    console.error("Could not find block in head");
    process.exit(1);
}

const usersBlock = headContent.substring(startIdx, endIdx).trim();

// Now we want to insert the conditional wrapper around usersBlock
const conditionalBlock = `{activeView === \x27users\x27 && (
  selectedUserKey ? (
    <UserManagement 
      userKey={selectedUserKey} 
      onBack={() => setSelectedUserKey(null)} 
    />
  ) : (
    ${usersBlock.substring("{activeView === \x27users\x27 && (".length, usersBlock.length - 1)}
  )
)}`;

const targetStr = "{activeView === \x27users\x27 && <UserManagement />}";
if (adminContent.includes(targetStr)) {
    adminContent = adminContent.replace(targetStr, conditionalBlock);
    fs.writeFileSync(adminPanelPath, adminContent);
    console.log("Successfully patched AdminPanel.tsx");
} else {
    console.error("Target string not found in AdminPanel.tsx");
}

File: scratch/fix_admin_users_view.py
import os

admin_path = "frontend/src/pages/AdminPanel.tsx"
head_path = "scratch/AdminPanel_head_utf8.tsx"

with open(admin_path, "r", encoding="utf-8") as f:
    admin_content = f.read()

with open(head_path, "r", encoding="utf-8") as f:
    head_content = f.read()

start_idx = head_content.find("{activeView === 'users' && (")
end_idx = head_content.find("{activeView === 'pending' && (")

if start_idx == -1 or end_idx == -1:
    print("Could not find block in head")
    exit(1)

users_block = head_content[start_idx:end_idx].strip()

# Strip the opening {activeView === 'users' && ( and closing )}
inner_content = users_block[len("{activeView === 'users' && ("):-2] # last two chars should be `)}`

conditional_block = """{activeView === 'users' && (
  selectedUserKey ? (
    <UserManagement 
      userKey={selectedUserKey} 
      onBack={() => setSelectedUserKey(null)} 
    />
  ) : (""" + inner_content + """)
)}"""

target_str = "{activeView === 'users' && <UserManagement />}"
if target_str in admin_content:
    admin_content = admin_content.replace(target_str, conditional_block)
    with open(admin_path, "w", encoding="utf-8") as f:
        f.write(admin_content)
    print("Successfully patched AdminPanel.tsx")
else:
    print("Target string not found in AdminPanel.tsx")
